src/core/trtserver.cc

```diff
@@ -1594,6 +1594,38 @@ TRTSERVER_ServerModelRepositoryIndex(
   return nullptr;  // success
 }
 
+// Overload to remove GRPC V2 dependency
+TRTSERVER_Error*
+TRTSERVER_ServerModelRepositoryIndexHTTP(
```

Comment:
New TRTSERVER API should be added in trtserver2.h. Call this one TRTSERVER2_ServerModelIndex, and see TRTSERVER_ServerExtensions for how to type the arguments. But I think you are actually going to have to create a TRTSERVER2_ModelIndex opaque object and return that (second argument will be TRTSERVER2_ModelIndex*, because otherwise you will not be able to control the life-cycle of the strings. See my comment below.